internal/converter/openapi_reconciler.go

```diff
@@ -24,6 +24,11 @@ type OpenAPIReconciler interface {
 	// goverter:ignore Id CreateTimeSinceEpoch LastUpdateTimeSinceEpoch Name RegisteredModelId
 	UpdateExistingModelVersion(source OpenapiUpdateWrapper[openapi.ModelVersion]) (openapi.ModelVersion, error)
 
+	// Ignore all fields that can't be updated
+	// goverter:default InitWithExisting
+	// goverter:autoMap Update
+	UpdateExistingArtifact(source OpenapiUpdateWrapper[openapi.Artifact]) (openapi.Artifact, error)
```

Comment:
I think the comment phrased in another way is possibly why differently than the other converter, this one seems to be missing the 

```go
// goverter:ignore Id CreateTimeSinceEpoch LastUpdateTimeSinceEpoch Name ArtifactType
```

?

Reply:
>this one seems to be missing the

I think that's because in this case the `Artifact` object, in according to how the model is auto-generated by openapi-generator, does not have the concrete fields (e.g., `id`..) but it does have `ModelArtifact` and `DocArtifact` and in according to which one is not `Nil` it will be of that type.

Given that, I think `UpdateExistingArtifact` should be somehow re-worked to consider this situation, which is different from all other entities we have right now.

Reply:
possibly a way to ignore the same set, whether they are in the DocArtifact or ModelArtifact with a dot notation?

not sure if that would work with the pointers tho

Reply:
we can simply create our own helper IMO. It would call into the appropriate update func depending on the artifact type, wdyt?

Reply:
sounds good with me, thanks @isinyaaa ; when you "wire it" in the comment, it would be helpful to "note down" this difference/motivation

Reply:
making a direct helper on the generator won't do, as `autoMap` won't take in custom mapping functions.
I'm going with a custom helper that takes in the reconciler to call the appropriate methods for each model.